base/boot.jl

```diff
@@ -60,15 +60,28 @@
 #end
 
 #type LambdaInfo
+#    inferred_ast::Expr
+#    rettype::Any
+#    sparam_vals::SimpleVector
+#    specTypes::Any
+#    unspecialized::LambdaInfo
+#    def::MethodInfo
+#    pure::Bool
```

Comment:
the `pure` field, as it exists currently, is essentially a placeholder for "ast properties that inference wants to stash somewhere"